app/cosi/server.go

```diff
@@ -0,0 +1,166 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"os/user"
+	"path"
+	"runtime"
+	"strings"
+
+	c "github.com/dedis/cothority/app/lib/config"
+	"github.com/dedis/cothority/log"
+	"github.com/dedis/cothority/network"
+	"gopkg.in/urfave/cli.v1"
+	// Empty imports to have the init-functions called which should
+	// register the protocol
+
+	_ "github.com/dedis/cothority/protocols/cosi"
+	_ "github.com/dedis/cothority/services/cosi"
+)
+
+func runServer(ctx *cli.Context) {
+	// first check the options
+	config := ctx.String("config")
+
+	if _, err := os.Stat(config); os.IsNotExist(err) {
+		log.Fatalf("[-] Configuration file does not exist. %s. "+
+			"Use `cosi server setup` to create one.", config)
+	}
+	// Let's read the config
+	_, conode, err := c.ParseCothorityd(config)
+	if err != nil {
+		log.Fatal("Couldn't parse config:", err)
+	}
+	conode.Start()
+}
+
+func stderr(format string, a ...interface{}) {
+	fmt.Fprintf(os.Stderr, fmt.Sprintf(format, a...)+"\n")
+}
+
+func stderrExit(format string, a ...interface{}) {
+	stderr(format, a...)
+	os.Exit(1)
+}
+
+func getDefaultConfigFile() string {
```

Comment:
Many of these functions are already implemented in `app/lib/{client,server}` if I remember correctly.